docs/api/interface_type.md

```diff
@@ -0,0 +1,119 @@
+
+# `GraphQLInterface`
+
+`GraphQLInterface` is a base class used to define GraphQL interfaces in a modular and reusable manner. It provides a structured way to create interfaces with fields, resolvers, and descriptions, making it easier to manage complex GraphQL schemas.
+
+## Inheritance
+
+- Inherits from `GraphQLBaseObject`.
+
+## Class Attributes
+
+- **`__schema__`**: *(Optional[str])* - The GraphQL schema definition string for the union, if provided.
+- **`__graphql_name__`**: *(Optional[str])* The custom name for the GraphQL type. If not provided, it will be generated based on the class name.
+- **`__description__`**: *(Optional[str])* A description of the GraphQL type, included in the schema documentation.
+- **`__graphql_type__`**: Defines the GraphQL class type, set to `GraphQLClassType.BASE`.
+- **`__aliases__`**: *(Optional[Dict[str, str]])* Defines field aliases for the object type.
+- **`__requires__`**: *(Optional[Iterable[Union[Type[GraphQLType], Type[Enum]]]])* Specifies other types or enums that are required by this object type.
+
+## Core Methods
+
+### `resolve_type`
+
+A static method that resolves the type of an object when using interfaces or unions. This method can be overridden to provide custom type resolution logic.
+
+### `resolver`
+
+Defines a resolver for a subscription. The resolver processes the data provided by the source before sending it to the client.
```

Comment:
Parts of this doc were copied from subscription doc but weren't updated after that.